File: Python Assignments/Python Assignment 4/Assignment4_2.py
#   2. Write a program which contains one lambda function which accepts
#      two parameters and return its multiplication.

MultiOfTwo = lambda A, B : A*B

def main():
    Number1 = int(input("Enter a first number : "))
    Number2 = int(input("Enter a second number : "))

    Result = MultiOfTwo(Number1,Number2)
    print("Multiplication of {0} and {1} is {2} ".format(Number1,Number2,Result))

if __name__ == "__main__":
    main()

#   Output:
#   D:\Python\Marvellous Python\Python Assignments\Python Assignment 4>Python Assignment4_2.p
#   Enter a first number : 4
#   Enter a second number : 3
#   Multiplication of 4 and 3 is 12

#   D:\Python\Marvellous Python\Python Assignments\Python Assignment 4Python Assignment4_2.py
#   Enter a first number : 6
#   Enter a second number : 3
#   Multiplication of 6 and 3 is 18

File: Logic_Building_Assignments/Logic_Building_Assignment_4/Assignment4_2.c
//  2. Write a program which accept number from user and display its
//     factors in decreasing order.

#include<stdio.h>

void FactRevOrder(int iNo)
{
    int iCnt = 0;

    if(iNo == 0)
    {
        printf("There are infinte factors of zero");
    }
    else if(iNo == 1)
    {
        printf("%d",iNo);
    }
    else
    {
        for(iCnt = (iNo/2); iCnt >= 1; iCnt--)
        {
            if((iNo % iCnt) == 0)
            {
                printf("%d\t",iCnt);
            }
        }
    }
}

int main()
{
    int iValue = 0;

    printf("Enter a number : \n");
    scanf("%d",&iValue);

    FactRevOrder(iValue);

    return 0;
}

/*

    Output:
    D:\Marvellous Infosystems\Logic Building with Industrial Project Batch\Logic Building Assignments\Logic Building Assignment 4>gcc Assignment4_2.c

    D:\Marvellous Infosystems\Logic Building with Industrial Project Batch\Logic Building Assignments\Logic Building Assignment 4>a
    Enter a number :
    12
    6       4       3       2       1
    D:\Marvellous Infosystems\Logic Building with Industrial Project Batch\Logic Building Assignments\Logic Building Assignment 4>a
    Enter a number :
    13
    1
    D:\Marvellous Infosystems\Logic Building with Industrial Project Batch\Logic Building Assignments\Logic Building Assignment 4>a
    Enter a number :
    10
    5       2       1

*/